File: base/src/main/java/com/inz/z/base/view/AbsBaseActivity.java
package com.inz.z.base.view;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @author Zhenglj
 * @version 1.0.0
 * Create by inz in 2019/05/23 15:36.
 */
public abstract class AbsBaseActivity extends AppCompatActivity {
    protected Context mContext;

    protected abstract void initWindow();

    @LayoutRes
    protected abstract int getLayoutId();

    protected abstract void initView();

    protected abstract void initData();

    /**
     * 是否使用 DataBinding , 返回true时，{@linkplain #setDataBindingView()} 必须复写
     *
     * @return 默认不使用
     */
    protected boolean useDataBinding() {
        return false;
    }

    /**
     * 设置DataBindingView 内容
     */
    protected void setDataBindingView() {

    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initWindow();
        if (!useDataBinding()) {
            setContentView(getLayoutId());
        } else {
            setDataBindingView();
        }
        mContext = this;
        initView();
        initData();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mContext = null;
    }
}

File: app_update/src/main/java/com/inz/z/app_update/view/AbsBaseActivity.kt
package com.inz.z.app_update.view

import android.content.Context
import android.os.Bundle
import android.support.annotation.IdRes
import android.support.annotation.LayoutRes
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity

/**
 *
 * @author Zhenglj
 * @version 1.0.0
 * Create by inz in 2019/3/13 15:07.
 */
abstract class AbsBaseActivity : AppCompatActivity() {

    protected var mContext: Context? = null

    protected abstract fun getUpdateDialogFragment(): Fragment

    protected abstract fun getDownloadDialogFragment(): Fragment

    @LayoutRes
    protected abstract fun getLayoutId(): Int

    protected fun initWindow() {

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initWindow()
        setContentView(getLayoutId())
        mContext = this
    }
}